### PR TITLE
Better leverage mathconstants module

### PR DESCRIPTION
**Pull request by hfp**
Improved calculating eval_transc_func. Adjusted complex_diag to deallocate work arrays prior to abort, and avoid unnecessary zeroing scratch-buffers on entry (mathlib.F and cp_cfm_diag.F).

### CODE REVIEW

File: src/common/mathlib.F [hunk 1]
```diff
@@ -1736,8 +1727,11 @@ ELEMENTAL SUBROUTINE eval_transc_func(r, eps, omega, fn, df)
       REAL(dp), INTENT(in)                               :: r, eps, omega
       REAL(dp), INTENT(out)                              :: fn, df
 
-         fn = erfc(omega*r) - r*eps
-         df = -omega*2*EXP(-omega**2*r**2)/SQRT(pi) - eps
+      REAL(dp)                                           :: qr
+
+         qr = omega*r
+         fn = ERFC(qr) - r*eps
+         df = -2.0_dp*oorootpi*omega*EXP(qr**2) - eps
```
Comment (mkrack) [inline src/common/mathlib.F]: It looks like a minus sign before `qr**2` has been lost here.

Comment (hfp) [reply src/common/mathlib.F]: Hmm, I thought the minus took precedence like negative number squared?

( Previous expression was `df = -omega*2*EXP(-omega**2*r**2)/SQRT(pi) - eps`. )

**Comment by hfp** [comment]
( Many references about Fortran's operator precedence document unary minus aka negate as taking precedence over exponentiation. )

**Comment by hfp** [comment]
( As stated in the previously (revised) description, this PR sneaked in a change of `complex_diag` and `cp_cfm_heevd`. )

**Pull request merged**